src/assets/src/components/common.tsx

```diff
@@ -440,6 +426,45 @@ export const LoginDialog = (props: LoginDialogProps) =>
         </Modal.Footer>
     </Modal>
 
+export interface DialogState {
+    show: boolean;
+    onClose?: () => void;
+    title?: string;
+    description?: string;
+    action?: () => void;
+}
+
+interface DialogProps extends DialogState {}
+
+export const Dialog = (props: DialogProps) => {
+    const { show, onClose, title, description, action } = props;
+
+    return (
+        <Modal
+            show={show}
+            onHide={onClose}
+        >
+            {(
+                props.title !== undefined &&
+                props.description !== undefined &&
+                props.action !== undefined
+            ) && (
+                <>
+                    <Modal.Header closeButton>
+                        <Modal.Title>{title}</Modal.Title>
+                    </Modal.Header>
+                    <Modal.Body>
+                        <p>{description}</p>
+                    </Modal.Body>
+                    <Modal.Footer>
+                        <Button onClick={props.onClose} variant="outline-dark">Cancel</Button>
+                        <Button onClick={action} variant="primary">OK</Button>
+                    </Modal.Footer>
+                </>
+            )}
+        </Modal>
+    );
```

Comment:
I see a pattern that lot of reusable components has been placed inside this `common.tsx` file. But I felt these very well be it's own components. Just an observation

Reply:
Yeah, it was a choice by the original frontend developers. I'm not prepared to change that at this point.